Add nuke-orphans plugin

sandorex opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/sandorex/nuke-orphans-plugin

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

liamcain commented:
https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/main.ts#L18 I think this can be simplified by using `Object.values` instead of `entries` in both instances, no?

https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/main.ts#L38 you shouldn't need to pass `this` as the second argument to `.filter` here

https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/main.ts#L46 `===`

https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/settings.ts#L49 IMO, it's better and simpler to just always read this option instead of providing your own setting. 

https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/settings.ts#L53
If you want to remove the `ts-ignore` here, I suggest you add a TS module override like [this](https://github.com/kometenstaub/linked-data-helper/blob/f0704cfda75e8a98ac29d52a7490a77053a41f8d/src/interfaces.ts#L32).

https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/trash_modal.ts#L44 You should read the `this.app.vault.getConfig("trashOption")` to see if the user wants deleted files in their local trash or in the Obsidian trash. (Also not a public API)

sandorex commented:
> https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/main.ts#L18 I think this can be simplified by using Object.values instead of entries in both instances, no?

Improved the readability significantly, thanks

> https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/main.ts#L38 you shouldn't need to pass this as the second argument to .filter here

Fixed

> https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/main.ts#L46 ===

Fixed

> https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/settings.ts#L49 IMO, it's better and simpler to just always read this option instead of providing your own setting.

Now reading `attachmentFolderPath` from config by default and allows overriding it

> https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/settings.ts#L53
If you want to remove the ts-ignore here, I suggest you add a TS module override like [this](https://github.com/kometenstaub/linked-data-helper/blob/f0704cfda75e8a98ac29d52a7490a77053a41f8d/src/interfaces.ts#L32).

Added module override

> https://github.com/sandorex/nuke-orphans-plugin/blob/044be34c22651521181406337ad539b795e35830/src/trash_modal.ts#L44 You should read the this.app.vault.getConfig("trashOption") to see if the user wants deleted files in their local trash or in the Obsidian trash. (Also not a public API)

Fixed
